src/ddsrt/include/dds/ddsrt/bswap.h

```diff
@@ -28,41 +28,41 @@ enum ddsrt_byte_order_selector {
   DDSRT_BOSEL_LE,
 };
 
+// coverity[ -tainted_data_return ]
```

Comment:
The root of the problem seems to me that Coverity considers byte-swapping a reason to treat the result as tainted, even though it is tainted if and only if the input was tainted. Often the heuristic works out, but sometimes we have to deal with byte-swapped trusted (= locally generated) data, and this annotations seems to throw out the valuable warnings as well.

I am not aware of a way to solve that problem short of having two sets of functions and annotating only one of them, but: (1) duplicating code like that is almost insane, and (2) it would do nothing to prevent accidentally using the one that strips the "taintedness" from the result (because of C's weak type system ...)

The only alternative that I know is explicitly marking the cases where it warns as "false positives" in the GUI. While annoying, I think that could be the better option. What do you think?

Reply:
I agree with both your points, though then it is more a "treating the symptoms rather than the root cause" approach, which I personally try to avoid whereever possible.

Though in this case I think your arguments for not causing false negatives in coverity weigh heavier than not solving the root cause of the reported issue.